server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -59,6 +56,7 @@ protected AbstractServer(String appName, ServerOpts opts, String[] args) {
    * Run this server in a main thread
    */
   public void runServer() throws Exception {
+    TraceUtil.enableServerTraces(getHostname(), applicationName, context.getConfiguration());
```

Comment:
This was moved from the constructor due to the hostname not being resolved at time abstract constructor was called.

Reply:
Probably fine... as long as the remaining body of the constructor doesn't try to trace anything.

Also, instead of moving this, we could try to make the setting of the hostname more standardized by moving it into the AbstractServer constructor itself.

Reply:
Yes.  It was very clean before when the address was more generic but using properties made it more difficult to manage in the absract server without having a long switch statement for each server type.  The property could use the "appname" which is already passed in as a prefix (I believe the naming convention is this already).   That actually may work..hmmm.

Reply:
The appname may work... but that might also be a user-configurable parameter, intended for metrics and/or logging configuration... so it might not consistently represent the configuration parameter name. A switch statement or another required method on the abstract server base class to provide the config prefix based on the server type might be needed.